models/base_model.py

```diff
@@ -41,11 +41,10 @@ def save(self):
 
     def to_dict(self):
         """Returns a dictionary representation of BaseModel instance"""
-        inst_dict = self.__dict__
+        inst_dict = self.__dict__.copy()
```

Comment:
To be honest I really don’t see how this changes anything as it’s the same dictionary object, just different addresses/offset, hopefully I do not understand the concept behind it and will need more info from you later…
Other than that…everything looks fine.